### PR TITLE
Added flyway and migration V1_0 and V1_1.

### PR DESCRIPTION
**Pull request by KrlSed**

### CODE REVIEW

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java [hunk 1]
```diff
@@ -33,13 +35,15 @@ public class Group {
   @GeneratedValue
   private UUID groupId;
 
-  @Column(nullable = false, unique = true)
+  @Column
   private String name;
 
+  @JsonIgnore
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: Why?

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java [hunk 2]
```diff
@@ -33,13 +35,15 @@ public class Group {
   @GeneratedValue
   private UUID groupId;
 
-  @Column(nullable = false, unique = true)
+  @Column
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: rollback

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java (hunk 1 above)
Comment (borecjudo2) [reply service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: and for other

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java (hunk 2 above)
Comment (KrlSed) [reply service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: Done

File: service/src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java [hunk 3]
```diff
@@ -24,7 +22,7 @@ public class GroupDto implements Serializable {
 
   private String name;
 
-  private Teacher teacher;
+  private UUID teacherId;
 
-  private Set<Student> students;
+  private Set<UUID> studentId;
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java]:
```suggestion
  private Set<UUID> studentIds;
```

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java [hunk 4]
```diff
@@ -36,10 +36,9 @@ public class Group {
...
-  @OneToOne
-  @JoinColumn(name = "teacher_id")
-  private Teacher teacher;
+  private UUID teacherId;
 
-  @OneToMany(mappedBy = "group")
-  private Set<Student> students;
+  @ElementCollection(targetClass = UUID.class)
+  @CollectionTable(name = "students", joinColumns = @JoinColumn(name = "group_id"))
+  private Set<UUID> studentId;
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]:
```suggestion
  private Set<UUID> studentIds;
```

File: service/src/test/java/com/sedliarov/learningtable/integration/RestIntegrationTestBase.java [hunk 5]
```diff
@@ -55,8 +55,8 @@ public void beforeEach() {
 
   public void clearDataBase() {
     studentRepository.deleteAll();
-    teacherRepository.deleteAll();
     groupRepository.deleteAll();
+    teacherRepository.deleteAll();
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/RestIntegrationTestBase.java:59]:
```suggestion
groupRepository.deleteAll();
studentRepository.deleteAll();
    teacherRepository.deleteAll();
```

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java [hunk 6]
```diff
@@ -184,7 +184,6 @@ void testUpdateTeacher() {
       softly.assertThat(updatedTeacher.getSecondName()).isEqualTo(teacherDtoForUpdate.getSecondName());
       softly.assertThat(updatedTeacher.getFirstName()).isEqualTo(teacherDtoForUpdate.getFirstName());
       softly.assertThat(updatedTeacher.isAdmin()).isEqualTo(teacherDtoForUpdate.isAdmin());
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java:186]: Add assertion for groupId

File: service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java [hunk 7]
```diff
@@ -20,7 +18,7 @@ public class GroupFixture {
 
   private static final String NAME = "0000000";
 
-  private static final Set<Student> EMPTY_SET = Collections.<Student>emptySet();
+  private static final Set<UUID> EMPTY_SET = Collections.<UUID>emptySet();
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]:
```suggestion
  private static final Set<UUID> EMPTY_SET = Collections.emptySet();
```

File: service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java [hunk 8]
```diff
@@ -47,49 +45,49 @@ public Group createEntityWithName(String name) {
   /**
    * Method for create Group entity and set teacher.
    */
-  public Group createEntityWithTeacher(Teacher teacher) {
+  public Group createEntityWithTeacher(UUID teacher) {
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]: And the same for other
```suggestion
  public Group createEntityWithTeacher(UUID teacherId) {
```

File: service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java [hunk 9]
```diff
@@ -47,49 +45,49 @@ public Group createEntityWithName(String name) {
...
-    group.setTeacher(teacher);
+    group.setTeacherId(teacher);
     return group;
   }
 
   /**
    * Method for create Group entity and set students.
    */
-  public Group createEntityWithStudents(Set<Student> students) {
+  public Group createEntityWithStudents(Set<UUID> students) {
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]: And the same for other
```suggestion
  public Group createEntityWithStudents(Set<UUID> studentIds) {
```

Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java (hunk 8 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java (hunk 7 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/utils/GroupFixture.java]: Done

File: service/src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java (hunk 3 above)
Comment (KrlSed) [reply service/src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java]: Done

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java (hunk 4 above)
Comment (KrlSed) [reply service/src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java (hunk 6 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java:186]: Done

**Review by borecjudo2 — APPROVED** [review]

**Pull request merged**